usr/lib/api/api_interface.c

```diff
@@ -2557,6 +2557,11 @@ CK_RV C_Initialize(CK_VOID_PTR pVoid)
     // Clear out the load list
     memset(slot_loaded, 0, sizeof(int) * NUMBER_SLOTS_MANAGED);
 
+    // Zero out API_Proc_Struct
+    // This must be done prior to all goto error calls, else bt_destroy()
+    // will fail because it accesses uninitialized memory when t->size > 0.
+    memset((char *) Anchor, 0, sizeof(API_Proc_Struct_t));
```

Comment:
is the cast to (char *) really needed?

Other than that the fix looks good. Thanks for reporting and fixing!

Reply:
Nope, the cast to `(char *)` is not necessary.

I left it because it was already there. It seems like there's a couple of other places where it is used (`utility.c`, `obj_mgr.c`, and `apiutil.c`).

Removed, thanks!